File: 프로그래머스/l2_가장 큰 수.py
def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key = lambda x: 3*x, reverse = True)
    result =''

    for n in numbers:
        result += n
    if int(result) == 0:
        return '0'
    else:
        return result

solution([3, 30, 34, 5, 9])












# def solution(numbers):
#     str_numbers = list(map(str, numbers))
#     str_numbers.sort(key = lambda x : 3*x, reverse=True)
#     result = ''
#     for n in str_numbers:
#         result +=n
#     if int(result) ==0:
#         return '0'
#     return result

File: 프로그래머스/JAVA/l2_가장큰수.java
import java.util.Arrays;
import java.util.Comparator;

public class l2_가장큰수 {
    public static void main(String[] args){
        int[] numbers = {3,30,34,5,9};
        String answer = solution(numbers);
    }

    public static String solution(int[] numbers){
        String[] nums = new String[numbers.length];
        for(int i=0; i<numbers.length; i++){
            nums[i] = numbers[i] +"";
        }
        System.out.println(Arrays.toString(nums));

        Arrays.sort(nums, new Comparator<String>() {
            @Override
            public int compare(String o1, String o2) {
                return (o2+o1).compareTo(o1+o2);
            }
        });

        String answer = "";
        for (String num:nums){
            answer += num;
        }
        return answer.charAt(0) =='0'? "0" : answer;

    }
}
